src/Workspaces/Core/Portable/Workspace/Solution/Project.cs

```diff
@@ -489,6 +489,33 @@ public Task<VersionStamp> GetDependentSemanticVersionAsync(CancellationToken can
         public Task<VersionStamp> GetSemanticVersionAsync(CancellationToken cancellationToken = default)
             => _projectState.GetSemanticVersionAsync(cancellationToken);
 
+        /// <summary>
+        /// Calculates a checksum that contains a project's checksum along with a checksum for each of the project's transitive dependencies.
+        /// </summary>
+        /// <remarks>
+        /// This checksum calculation is used to determine if a diagnostics need to be recalculated based on the last reported checksum.
```

Comment:
might need to reword slightly so it doesn't reference diagnostics

Reply:
yup!